src/services/upper-bar-manager/models/icon-model.ts

```diff
@@ -27,6 +28,7 @@ export class IconModel {
     this.componentRef = createRef();
     this.presets =
       item.presets && item.presets.length > 0 ? item.presets : [ReservedPresetNames.Playback, ReservedPresetNames.Live];
+    this.shouldHandleOnClick = typeof item.shouldHandleOnClick === 'boolean' ? item.shouldHandleOnClick : true;
```

Comment:
why do we need this ?  why just check if on click is supplied - call it

Reply:
the new `shouldHandleOnClick` flag is indicating whether the upper bar should call the icon's `onClick` or not.
the reason is that the controls i handled to move to the upper bar (PiP, CC, AAD, etc..) are not only icons like the vamb plugin icons- they are actually the component itself; the way they behave in bottom bar- same in upper bar. which means, they have their own states and handlers.

**another explanation:**
let's say we don't have that flag, then the `onClick` will happen **twice**- 1 time from within the component and 1 time from the upper bar. Thus, the solution for this issue was to tell the upper bar- "I can handle myself, don't trigger my onClick".